### PR TITLE
Migrate deployment from broken Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by jchodera**
This PR migrates hugo static site builds and deployment from Travis CI (which appears to be defunct) to GitHub Actions.

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          extended: true
+
+      - name: Build
+        run: hugo --minify
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./public
```
Comment (mikemhenry) [inline .github/workflows/gh-pages.yml]:
```suggestion
          publish_dir: ./public
          cname: openforcefield.org
```
I think we need this

Comment (mikemhenry) [reply .github/workflows/gh-pages.yml]: We need a way to test this since if we YOLO it we might take the website down. I will investigate.

**Comment by mikemhenry** [comment]
So it looks like the best way to test is this build on a different branch, then in the web UI switch over to the new branch and see if it looks okay and if everything is broken, we can switch it back to the old branch.

We might want to do this at night or something so we can minimize disruptions

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (jchodera) [reply .github/workflows/gh-pages.yml]: Thanks!

**Comment by jchodera** [comment]
@mikemhenry : Would you be willing to give this a go? I've just given you Admin access!

**Comment by mikemhenry** [comment]
@jchodera Yes! I think I will wait until this evening to not be too disruptive

**Comment by mikemhenry** [comment]
I got this working now + added branch protection so updates require a PR

**Pull request closed without merging**